apps/api/src/app/shared/framework/response.decorator.ts

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { ApiExtraModels, ApiOkResponse, ApiCreatedResponse, getSchemaPath } from '@nestjs/swagger';
+import { Type, applyDecorators } from '@nestjs/common';
+import { DataWrapperDto } from '../dtos/data-wrapper-dto';
+
+export const ApiResponse = <DataDto extends Type<unknown>>(
+  dataDto: DataDto,
+  type: 200 | 201 = 200,
```

Comment:
can we call this statusCode, makes it easier to use from other files to understands what it is :)